File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        def reverse(x: int) -> int:
            res = 0
            while x > 0:
                res = res * 10 + (x % 10)
                x //= 10
            return res
        res = len(nums)
        pos = {}
        pos[nums[-1]] = len(nums) - 1
        for i in range(len(nums) - 2, -1, -1):
            rev = reverse(nums[i])
            if rev in pos:
                res = min(res, pos[rev] - i)
            pos[nums[i]] = i
        return -1 if res == len(nums) else res

File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.cpp
class Solution {
public:
    int minMirrorPairDistance(vector<int>& nums) {
        int n = nums.size();
        int res = n;
        unordered_map<int, int> pos;
        pos[nums[n - 1]] = n - 1;
        for(int i = n - 2; i >= 0; i--)
        {
            int rev = reverse(nums[i]);
            cout << rev << endl;
            if(pos.count(rev))
            {
                res = min(res, pos[rev] - i);
            }
            pos[nums[i]] = i;
        }
        return res == n ? -1 : res;
    }
private:
    int reverse(int x)
    {
        int res = 0;
        while(x > 0)
        {
            res = res * 10 + (x % 10);
            x /= 10;
        }
        return res;
    }
};